cvat/apps/engine/views.py

```diff
@@ -1070,22 +1051,55 @@ def upload_finished(self, request):
                 # if the value of stop_frame is 0, then inside the function we cannot know
                 # the value specified by the user or it's default value from the database
                 data['stop_frame'] = None
-            task.create(self._object, data, request)
-            return Response(serializer.data, status=status.HTTP_202_ACCEPTED)
+
+        # Need to process task data when the transaction is committed
+        task.create(self._object, data, request)
+
+        return Response(serializer.data, status=status.HTTP_202_ACCEPTED)
+
+    @transaction.atomic
+    def _handle_upload_backup(self, request):
```

Comment:
IMO, `_handle_upload_annotations`, `_handle_upload_data`, `_handle_upload_backup` should be moved closer to the function where thery're used (`upload_finished`). That would make the code easier to read (and the diff too).